### PR TITLE
Use GitHub Actions cache backend for Docker

### PR DESCRIPTION
**Pull request by divergentdave**
This enables use of the GitHub Actions cache to cache Docker image layers for the conformance test. When the new environment variable `DNS_TEST_DOCKER_CACHE_GHA` is set, extra command line flags will be passed to `docker build`. This cache backend also requires using the `docker-container` build driver, so that is set up using the `docker/setup-buildx-action` workflow. I also made it required to explicitly choose a Hickory DNS DNSSEC feature when running conformance tests, to eliminate some untested edge cases and inconsistent behavior when no feature is selected.

### CODE REVIEW

File: .github/workflows/conformance.yml [hunk 1]
```diff
@@ -13,13 +13,19 @@ jobs:
...
+    env:
+      DOCKER_BUILDKIT: 1
     steps:
       - uses: actions/checkout@v4
       - uses: dtolnay/rust-toolchain@master
         with:
           toolchain: stable
           components: clippy, rustfmt
       - uses: extractions/setup-just@v3
+      - uses: docker/setup-buildx-action@v3
```
Comment (djc) [inline .github/workflows/conformance.yml:26]: Why is this useful/what does it fix? Maybe add 1-2 lines of comments on the rationale here?

File: conformance/packages/dns-test/src/container.rs [hunk 2]
```diff
@@ -329,6 +362,10 @@ fn skip_docker_build() -> bool {
     env::var("DNS_TEST_SKIP_DOCKER_BUILD").is_ok()
 }
 
+fn docker_build_gha_cache() -> bool {
+    env::var("DNS_TEST_DOCKER_CACHE_GHA").is_ok()
```
Comment (djc) [inline conformance/packages/dns-test/src/container.rs:351]: Maybe [`env!()`](https://doc.rust-lang.org/std/macro.env.html)?

File: conformance/packages/dns-test/src/lib.rs [hunk 3]
```diff
@@ -130,20 +130,18 @@ fn parse_implementation(env_var: &str) -> Implementation {
             return Implementation::Bind;
         }
 
-        if subject.starts_with("hickory") {
-            let Some(rest) = subject.strip_prefix("hickory ") else {
+        if subject.starts_with("hickory ") {
+            let tokens = subject.split_ascii_whitespace().collect::<Vec<_>>();
+            let Ok([_, url, dnssec_feature]) = TryInto::<[&str; 3]>::try_into(tokens) else {
```
Comment (djc) [inline conformance/packages/dns-test/src/lib.rs:135]: Nit: maybe `<[&str; 3]>::try_from(tokens)`?

**Review by djc — APPROVED** [review]
Have you already measured how much of an improvement this is, or do we need to merge it first for that to work?

File: conformance/packages/dns-test/src/container.rs (hunk 2 above)
Comment (divergentdave) [reply conformance/packages/dns-test/src/container.rs:351]: I think I'd rather keep it at run-time to keep it consistent with the other `DNS_TEST` environment variables.

**Comment by divergentdave** [comment]
This isn't working yet, I'm seeing `building with "default" instance using docker driver` when `docker build` starts. This isn't the driver I wanted to use, so it's preventing cache entries from being uploaded.

**Comment by divergentdave** [comment]
Looks like it cuts time by a few minutes in the good case, and adds a few minutes when it has to rebuild everything. Switching to the `docker-container` driver adds some time in the "exporting to docker image format", "importing to docker", and "exporting to docker image format" steps. However, that is more than made up by the cache hits.

* Before: 26m 37s https://github.com/hickory-dns/hickory-dns/actions/runs/13972947733/job/39120900389
* After, cold cache: 30m 28s https://github.com/hickory-dns/hickory-dns/actions/runs/13978493400/job/39138107424
* After, warm cache: 21m 23s https://github.com/hickory-dns/hickory-dns/actions/runs/13979005906/job/39142122361
